packages/web/server/api/edge-routers/one-click-trading-router.ts

```diff
@@ -40,6 +44,49 @@ export const oneClickTradingRouter = createTRPCRouter({
   getNetworkFeeLimitStep: publicProcedure.query(async () =>
     getNetworkFeeLimitStep()
   ),
+  getSessionAuthenticator: publicProcedure
+    .input(z.object({ userOsmoAddress: z.string(), publicKey: z.string() }))
+    .query(async ({ input }) => {
+      const authenticators = await getAuthenticators({
+        userOsmoAddress: input.userOsmoAddress,
+      });
+
+      const subAuthenticators = authenticators
+        .filter(
+          (authenticator): authenticator is AllOfAuthenticator =>
+            authenticator.type === "AllOfAuthenticator"
+        )
+        .flatMap((authenticator) =>
+          authenticator.subAuthenticators.map((sub) => ({
+            ...sub,
+            authenticatorId: authenticator.id,
+          }))
+        );
+
+      if (subAuthenticators.length === 0) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Session not found: authenticators array is empty",
+        });
```

Comment:
Do we need to throw TRPCError objects ourselves, or will tRPC convert normal errors into those objects?

Reply:
Yes, since TRPC uses these errors to determine the query's status. Normal errors always default to a status of `500`